internal/provider/serverless_cluster.go

```diff
@@ -0,0 +1,168 @@
+package provider
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/scylladb/terraform-provider-scylladbcloud/internal/scylla"
+	"github.com/scylladb/terraform-provider-scylladbcloud/internal/scylla/model"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	serverlessClusterRetryTimeout    = 40 * time.Minute
+	serverlessClusterDeleteTimeout   = 90 * time.Minute
+	serverlessClusterRetryDelay      = 5 * time.Second
+	serverlessClusterRetryMinTimeout = 15 * time.Second
+	serverlessClusterPollInterval    = 10 * time.Second
+)
+
+func ResourceServerlessCluster() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceServerlessClusterCreate,
+		ReadContext:   resourceServerlessClusterRead,
+		UpdateContext: resourceServerlessClusterUpdate,
+		DeleteContext: resourceServerlessClusterDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(serverlessClusterRetryTimeout),
+			Update: schema.DefaultTimeout(serverlessClusterRetryTimeout),
+			Delete: schema.DefaultTimeout(serverlessClusterDeleteTimeout),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
+				Description: "Serverless cluster id",
+				Computed:    true,
+				Type:        schema.TypeInt,
+			},
+			"name": {
+				Description: "Serverless cluster name",
+				Required:    true,
+				ForceNew:    true,
+				Type:        schema.TypeString,
+			},
+			"free_tier": {
+				Description: "Whether a cluster is in a free tier",
+				Computed:    true,
+				Type:        schema.TypeBool,
+			},
+		},
+	}
+}
+
+func resourceServerlessClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		c = meta.(*scylla.Client)
+		r = &model.ClusterCreateRequest{
```

Comment:
those fields below are not configurable, or this is some temporary stuff?

Reply:
Currently the name is configurable for serverless clusters, the other fields happen to be the required defaults.